src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -203,9 +192,9 @@ public string GenerateEndTag()
         {
             if (SelfClosing || _isTagNameNullOrWhitespace)
             {
-                return string.Empty;
+                return null;
```

Comment:
you haven't changed this method to return an `ITextWriterCopyable`.  so why return `null`?